extensions/positron-connections/src/drivers.ts

```diff
@@ -7,12 +7,107 @@ import * as positron from 'positron';
 import * as vscode from 'vscode';
 
 export function registerConnectionDrivers(context: vscode.ExtensionContext) {
-	context.subscriptions.push(
-		positron.connections.registerConnectionDriver(new RPostgreSQLDriver())
-	);
+	for (const driver of [new RSQLiteDriver(), new RPostgreSQLDriver(), new PythonSQLiteDriver()]) {
+		context.subscriptions.push(
+			positron.connections.registerConnectionDriver(driver)
+		);
+	}
+}
+
+///  A generic driver implementation
+class RDriver implements positron.ConnectionsDriver {
+
+	driverId: string = 'unknown';
+	metadata: positron.ConnectionsDriverMetadata = {
+		languageId: 'r',
+		name: 'Unknown',
+		inputs: []
+	};
+
+	constructor(readonly packages: string[]) { }
+
+	async connect(code: string) {
+		const exec = await positron.runtime.executeCode(
+			'r',
+			code,
+			true,
+			false,
+			positron.RuntimeCodeExecutionMode.Interactive,
+			positron.RuntimeErrorBehavior.Continue
+		);
+		if (!exec) {
+			throw new Error('Failed to execute code');
+		}
+		return;
+	}
+
+	async checkDependencies() {
+		// Currently we skip dependency checks if there's no active R session
+		// in the foreground.
+		if (this.packages.length === 0) {
+			return true;
+		}
+
+		const session = await positron.runtime.getForegroundSession();
+		if (session) {
+
+			if (session.runtimeMetadata.languageId !== 'r') {
+				return true;
+			}
+
+			for (const pkg of this.packages) {
+				const installed = await session.callMethod?.('is_installed', pkg);
```

Comment:
This needs some exception handling; if `callMethod` throws I think the exception should be logged (and we should probably presume the dependencies to be installed since otherwise the user will be blocked as I understand it?)